twitter_analysis/app.py

```diff
@@ -0,0 +1,166 @@
+from flask import Flask, render_template
+import os
+from pymongo import MongoClient
+import requests
+import csv
+from decouple import config
+from datetime import datetime
+import pickle
+from xgboost import XGBClassifier
+import numpy as np
+from collections import deque
+import pandas as pd 
+from pandas import MultiIndex, Int16Dtype
+import json
+app = Flask(__name__)
+
+
+class TweetCollection:
+    def __init__(self):
+        f = open('topics.json')
+        self.query_list = deque(json.load(f))
+
+    def get_tweets(self):
+        while True:
+            self.get_all_tweets()
+
+    def get_all_tweets(self):
+        query_params = ''
+        TWITTER_API_1 = config('TWITTER_API_1')
+        q_obj = self.query_list.popleft()
+        if 'next_results' not in q_obj.keys():
+            query_params = "?q="+q_obj['query']+" AND -filter:retweets&country_code=US&result_type=recent&count=100&lang=en&tweet_mode=extended"
+        else:
+            query_params = q_obj['next_results']
+        headers = {'Authorization': config('API_TOKEN')}
+        api_url = TWITTER_API_1 + query_params
+        cluster= MongoClient(config('MONGO_URL'))
+        db = cluster['TweetMind']
+        collection = db['tweets_data']
+        tweets = requests.get(api_url,headers=headers).json()
+        if 'next_results' in tweets['search_metadata'].keys():
+            param = tweets['search_metadata']['next_results']
+            q_obj['next_results'] = param +'&tweet_mode=extended'
+        self.query_list.append(q_obj)
+        count = 0
+        for i in tweets['statuses']:
```

Comment:
this loop is likely a function.
eg: for tweet in tweets: 
user_features = getKeyFeaturesFromTweet(tweet)

def getKeyFeaturesFromTweet(tweet): 
 account_age_days = (datetime.now() - datetime.strptime(user['created_at'],'%a %b %d %X %z %Y').replace(tzinfo=None)).days
            verified = user['verified']
            etc....
return user_features